src/handlers/proposalsHandlers.ts

```diff
@@ -56,7 +62,175 @@ const getProposals = (
     });
 };
 
-const getProposal = (
+/**
+ * Merges the given field values into a new copy of the given proposal.
+ * It is OK if there are no versions or zero field values.
+ * @param proposal The proposal with versions but no values.
+ * @param values The values to merge into the given proposal.
+ * @returns A new proposal with the values merged into their respective versions.
+ */
+export const mergeProposalFieldValues = (
+  proposal: Proposal,
+  values: ProposalFieldValue[],
+): Proposal => {
+  const newProposal = structuredClone(proposal);
+  const { versions } = newProposal;
+  if (versions === undefined) {
+    // In this case there are no versions associated, return a simple copy.
+    return newProposal;
+  }
+  // Because both queries are sorted by proposal version, we don't need to do N*M loops.
```

Comment:
This function can't make assumptions about its input without validating those assumptions -- it has to be able to handle un-sorted input (or error if the inputs aren't sorted).  This could mean running an O(n) sort on each before hitting this code block though!

On a related note though we should avoid needing this kind of comment; I'm not quite sure what it means and it would be great to craft an algorithm that is self explanatory

Reply:
Fair enough, I see a place where it unsorted versions could slip through below, after looking more carefully. The comment should be clearer too. For example it can describe why it does what it does rather than talking about what it might have done.

Reply:
I think this one becomes moot by the refactoring in c602b6a.